src/Core/Form/IdentifiableObject/CommandBuilder/Product/SEOCommandsBuilder.php

```diff
@@ -61,6 +63,20 @@ public function buildCommands(ProductId $productId, array $formData): array
             $command->setRedirectOption($redirectionData['type'], $targetId);
         }
 
-        return [$command];
+        if (!empty($seoData['tags'])) {
+            $parsedTags = [];
+            if (is_array($seoData['tags'])) {
```

Comment:
@saulaski @Progi1984 this is already handled here which is more efficient because it allows detecting invalid values and throw an exception (see the builder tests for invalid case and more details)

Reply:
Is it the exception message ?
`Empty array of product tags provided in %s. To remove all product tags use %s.`
If so the message doesn't sound really explicit. Here the main issue is the `tags` option type expected to be an array of string tags for each language.
IMHO failure should get detected earlier at parsing level. Easier to debug.
But I'm being picky here, so I approve :)

Reply:
Yeah, you're right! The exception thrown was not related to the use case I was intending to check It was only a side effect Besides it made me realize the `RemoveAllProductTagsCommand` was never handled by the builder (even though it actually works by setting a localized array with empty values..)